test/pages/Index.spec.js

```diff
@@ -82,4 +140,58 @@ describe('Index', () => {
       )
     })
   })
+
+  // TODO: Again would be better with an integration test
+  // I also think this the complexity of this test is a bit of a code smell. Defo needs a refactor
+  describe('#deleteReports', () => {
+    let rows, reportFilenames, lessReportsJson
+
+    beforeEach(() => {
+      axios.post = jest.fn()
+      reportFilenames = ReportsJson.reports
+        .map((report) => report.filename)
+        .slice(0, 3)
+      lessReportsJson = { reports: ReportsJson.reports.slice(3, 5) }
+      $axios.$get.mockResolvedValue(ReportsJson)
+    })
+
+    it('when the request is successful', async () => {
+      axios.post.mockResolvedValue('success')
+      const wrapper = mount(Index, { localVue, mocks: { $axios, process } })
+      await flushPromises()
+      rows = wrapper.find('tbody').findAll('tr')
+      const arr = [0, 1, 2]
+      arr.forEach((i) => {
+        rows
+          .at(i)
+          .find('.selected input[type="checkbox"]')
+          .setChecked(true)
+      })
+      expect(wrapper.vm.reportsToDelete).toEqual(reportFilenames)
+      $axios.$get.mockResolvedValue(lessReportsJson)
+      const button = wrapper.find('#deleteReports')
+      await button.trigger('click')
+      await flushPromises()
```

Comment:
The await is when you delete the reports and the flushPromises is for when the reports are requested so it's not because I am flushing a promise I don't know about. You can't await it because it is the provider function which is triggered internally. The other flushPromises are the same. The reports are requested when the wrapper is mounted and you can't await a mount.

Reply:
Ah so maybe instead the provider function or refreshTable could be mocked out  - this comment might be outdated now though